### DISCUSSION TITLE
Question on data imputation for well data

### DISCUSSION THREAD

**Discussion by ivan-marroquin** [amices/mice#370] [Missing data methodology]
Hi,

Thanks for making such great tool available to us!

I am new into this type of analysis and I would like to know if mice can be applied to my data. I work with logging data (continuous and discrete). The missing information can be caused by geologic conditions, drilling operations, and/or logging costs. Since these conditions are external to actual data measurements, I believe the missing data follow a MCAR mechanism. 

However, the number of data points from one well to the next may vary, and this causes vectors of unequal length. In the sketch (see zip file), I show this issue. As you can see from the sketch, the missing data is towards the tail of the recorded data. Given such characteristic, is still possible to apply mice to replace missing values?

A second question is about null values are not replaced by imputed values. To show this, I ran a little test (see input data in zip file):

```
library(mice)

# Set column names for input data
well_names <- c('Well_A', 'Well_B', 'Well_C', 'Well_D', 'Well_E', 'Well_F', 'Well_G', 'Well_H')
				
# Read input data
litho <- read.csv("E:\\problematic_upscaled_lithofacies_data.csv", skip= 0, header= FALSE, col.names= well_names, na.strings= "NA")

# Generate data frame for imputation analysis
X <- litho
X[is.na(X)] <- NA
X <- as.data.frame(lapply(X, as.factor))

impData <- mice(X, m= 5, maxit= 50, method= 'polyreg', nnet.maxit= 100, nnet.trace= FALSE, nnet.MaxNWts= 1500, seed= 1969)
```

The print out of impData$imp shows that for 'Well_G' the null values were not replaced. Is this result due to the amount of discrete data versus the number of null entries? If null entries is above a certain threshold, then the null values are not replaced. Am I correct?

Many thanks for your help,

Ivan

[Missing_data_distribution.zip](https://github.com/stefvanbuuren/mice/files/4489161/Missing_data_distribution.zip)

**Comment by stefvanbuuren** [comment]
Ivan, 

Question 1: This is not the appropriate place to pose methodological questions. Rather direct to stack overflow. 
Question 2: Well_G is not imputed because it is constant. `mice` detects this, and removes it from the imputation process.

Hope this helps, Stef.

**Comment by ivan-marroquin** [comment]
Hi @stefvanbuuren ,

thanks so much for the quick response.

Ivan
